docs/src/migrate.md

```diff
@@ -0,0 +1,236 @@
+# Migration guides
+
+This guide provides an overview of how to migrate your code from either the pre-1.0 JSON.jl package to the 1.0 release or from JSON3.jl. The 1.0 release introduces several improvements and changes, particularly in how JSON is read and written, leveraging StructUtils.jl for customization and extensibility. Below, we outline the key differences and provide step-by-step instructions for updating your code.
+
+---
+
+## Migration guide from pre-1.0 -> 1.0
+
+### Writing JSON
+- `JSON.json`
+  - What stayed the same:
+    - Produces a compact String by default
+    - Can automatically serialize basic structs in a sensible way
+    - Can take an integer 2nd argument to induce "pretty printing" of the JSON output
+  - What changed:
+    - Can now pass `JSON.json(x; pretty=true)` or `JSON.json(x; pretty=4)` to control pretty printing
+    - Can pass filename as first argument to write JSON directly to a file `JSON.json(file, x)` the file name is returned
+    - Can pass any `IO` as 1st argument to write JSON to it: `JSON.json(io, x)`
+    - Circular reference tracking is fixed/improved (previously peer references were written as null)
+    - Explicit keyword arguments to control a number of serialization features, including:
+      - `omit_null::Bool` whether `nothing`/`missing` Julia values should be skipped when serializing
+      - `omit_empty::Bool` whether empty Julia collection values should be skipped when serializing
+      - `allownan::Bool` similar to the parsing keyword argument to allow/disallow writing of invalid JSON values `NaN`, `-Inf`, and `Inf`
+      - `ninf::String` the string to write if `allownan=true` and serializing `-Inf`
+      - `inf::String` the string to write if `allownan=true` and serializing `Inf`
```

Comment:
possibly `inf_string`? Also do we need `ninf` or can it just print as a `-` before `inf`?